addons/mail/models/mail_channel.py

```diff
@@ -265,7 +265,13 @@ def _alias_get_creation_values(self):
 
     def _subscribe_users(self):
         for mail_channel in self:
+            partners_to_add = mail_channel.group_ids.users.partner_id - mail_channel.channel_partner_ids
             mail_channel.write({'channel_partner_ids': [(4, pid) for pid in mail_channel.mapped('group_ids').mapped('users').mapped('partner_id').ids]})
+            if partners_to_add and len(partners_to_add) > 0:
+                self.env['mail.channel.partner'].search([
+                    ('channel_id', '=', mail_channel.id),
+                    ('partner_id', 'in', partners_to_add.ids)
+                ]).write({'is_pinned': True})
```

Comment:
Don't you need to use a search to determine the channel partner that needs to be created ?

Reply:
Do you really need to search for it though?

Once you have the partners to add (as per current diff), you just have to `create` the corresponding channel partner in the correct channel. This is just a `create`. Since we'll create the proper way, it will have default value, and we won't need to search or write it again.

Reply:
Also you can build up a list of channel.partner to create, and call it only once at the end of the method, outside of the loop, to fully batch the create.